File: app/src/main/java/c/m/storyapp/authentication_check/domain/use_case/get_token_from_data_store_use_case/GetTokenFromDataStoreUseCase.kt
package c.m.storyapp.authentication_check.domain.use_case.get_token_from_data_store_use_case

import c.m.storyapp.authentication_check.domain.model.AuthenticationToken
import c.m.storyapp.authentication_check.domain.repository.AuthenticationCheckRepository
import c.m.storyapp.common.util.Resource
import kotlinx.coroutines.flow.Flow
import javax.inject.Inject

class GetTokenFromDataStoreUseCase @Inject constructor(private val authenticationCheckRepository: AuthenticationCheckRepository) {
    suspend operator fun invoke(): Flow<Resource<AuthenticationToken>> {
        return authenticationCheckRepository.getToken()
    }
}

File: lib/authentication_check/domain/use_case/get_token_from_data_store_use_case/get_token_from_data_store_use_case.dart
import 'package:dartz/dartz.dart';
import 'package:injectable/injectable.dart';
import 'package:story_app/authentication_check/domain/model/authentication_token.dart';
import 'package:story_app/authentication_check/domain/repository/authentication_check_repository.dart';
import 'package:story_app/common/util/failure/failure.dart';
import 'package:story_app/common/util/failure/server_failure.dart';
import 'package:story_app/common/util/resource.dart';

@injectable
class GetTokenFromDataStoreUseCase {
  final AuthenticationCheckRepository _authenticationCheckRepository;

  GetTokenFromDataStoreUseCase(this._authenticationCheckRepository);

  Future<Either<Failure, AuthenticationToken>> call() async {
    final resource = await _authenticationCheckRepository.getToken();

    if (resource is Success) {
      return Right(resource.data ?? const AuthenticationToken(token: ''));
    } else {
      return Left(ServerFailure(message: resource.message));
    }
  }
}
